airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/assets/registry.py

```diff
@@ -148,70 +160,150 @@ def construct_registry_with_spec_from_registry(registry: dict, cached_specs: Out
+    return file_handle
+
+
+# New Registry
+
+
+@asset(required_resource_keys={"registry_directory_manager"}, group_name=GROUP_NAME)
 def cloud_registry_from_metadata(
-    registry_derived_metadata_definitions: List[PartialMetadataDefinition], cached_specs: OutputDataFrame
-) -> dict:
+    context: OpExecutionContext, metadata_definitions: List[MetadataDefinition], cached_specs: OutputDataFrame
+) -> Output[ConnectorRegistryV0]:
     """
     This asset is used to generate the cloud registry from the metadata definitions.
-
-    TODO (ben): This asset should be updated to use the GCS metadata definitions once available.
     """
-    from_metadata = construct_registry_from_metadata(registry_derived_metadata_definitions, "cloud")
-    from_metadata_and_spec = construct_registry_with_spec_from_registry(from_metadata, cached_specs)
-    return from_metadata_and_spec
+    registry_name = "cloud"
+    registry_directory_manager = context.resources.registry_directory_manager
 
+    from_metadata = construct_registry_from_metadata(metadata_definitions, registry_name)
+    registry_dict = construct_registry_with_spec_from_registry(from_metadata, cached_specs)
+    registry_model = ConnectorRegistryV0.parse_obj(registry_dict)
 
-@asset(group_name=GROUP_NAME)
-def oss_registry_from_metadata(registry_derived_metadata_definitions: List[PartialMetadataDefinition], cached_specs: OutputDataFrame) -> dict:
+    file_handle = persist_registry_to_json(registry_model, registry_name, registry_directory_manager)
+
+    metadata = {
+        "gcs_path": MetadataValue.url(file_handle.gcs_path),
+    }
+
+    return Output(metadata=metadata, value=registry_model)
+
+
+@asset(required_resource_keys={"registry_directory_manager"}, group_name=GROUP_NAME)
+def oss_registry_from_metadata(
+    context: OpExecutionContext, metadata_definitions: List[MetadataDefinition], cached_specs: OutputDataFrame
+) -> Output[ConnectorRegistryV0]:
     """
     This asset is used to generate the oss registry from the metadata definitions.
-
-    TODO (ben): This asset should be updated to use the GCS metadata definitions once available.
     """
-    from_metadata = construct_registry_from_metadata(registry_derived_metadata_definitions, "oss")
-    from_metadata_and_spec = construct_registry_with_spec_from_registry(from_metadata, cached_specs)
-    return from_metadata_and_spec
+    registry_name = "oss"
+    registry_directory_manager = context.resources.registry_directory_manager
+
+    from_metadata = construct_registry_from_metadata(metadata_definitions, registry_name)
+    registry_dict = construct_registry_with_spec_from_registry(from_metadata, cached_specs)
+    registry_model = ConnectorRegistryV0.parse_obj(registry_dict)
+
+    file_handle = persist_registry_to_json(registry_model, registry_name, registry_directory_manager)
+
+    metadata = {
+        "gcs_path": MetadataValue.url(file_handle.gcs_path),
+    }
+
+    return Output(metadata=metadata, value=registry_model)
```

Comment:
Lots of repeated code in between these two functions. Could you introduce an helper function to capture the common logic?